### PR TITLE
Enable GRPC_ERROR_IS_ABSEIL_STATUS

### PR DESCRIPTION
**Pull request by veblush**
Roll-forward #28898

### CODE REVIEW

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://fusion2.corp.google.com/invocations/b58a262f-55e9-4d88-acdb-9da440ab7171/targets): running

File: include/grpc/impl/codegen/port_platform.h
Comment (ctiller) [inline include/grpc/impl/codegen/port_platform.h:54]: Shouldn't we just remove this?

Comment (veblush) [reply include/grpc/impl/codegen/port_platform.h:54]: That will be done sometime after this change is proven safe.

Comment (ctiller) [reply include/grpc/impl/codegen/port_platform.h:54]: Should we leave the configuration in BUILD then, to be able to quickly toggle it?

**Comment by ctiller** [comment]
Noting the large bloat increase: this change is likely to be requested rolled back due to a size increase on mobile platforms.

File: include/grpc/impl/codegen/port_platform.h
Comment (veblush) [reply include/grpc/impl/codegen/port_platform.h]: Yeah, that's a good idea. I just reintroduced it again.

**Comment by veblush** [comment]
> Noting the large bloat increase: this change is likely to be requested rolled back due to a size increase on mobile platforms.

Good point. And I briefly checked with the internal iOS sample app. (OSSDemo.app) 
Build without and without this change;
- 17,333,514 bytes
- 17,596,426 bytes (1.5% increase)

Actual size increase is smaller than 8.0% so it'd be okay to have it. But it's worth investigation.

**Comment by veblush** [comment]
From 
- [Bloaty test result](https://gist.github.com/veblush/c481995bf4a951645994debd4a3c7279)
- [Bloaty test result with -fno-exceptions](https://gist.github.com/veblush/359abe14c9ce60fa2dd35391095502ef)

It seems that having C++ object instead of the pointer to the struct inherently increases the code size because
- It needs to generate C++ exception stack unwinding because of `absl::Status`. Once exception is disabled (with -fno-exceptions), size increase decreased from +383Ki to +261Ki.
- Checking if error is none adds more code for `absl::Status` because it needs to create OkStatus() object and call Equal function. This overhead will be mitigated by https://github.com/grpc/grpc/pull/29937.

**Review by markdroth — APPROVED** [review]
LGTM, but please get Craig's approval as well before merging.

**Review by ctiller — APPROVED** [review]
Approved on the basis that there'll be a cherry pick in that we can roll back separately to the normal flow: I expect some pushback on the bloat issue.

**Pull request merged**
